### PR TITLE
tag support multi level

### PR DESCRIPTION
**Pull request by carlvine500**
## What is the purpose of the change
fix issue #12672

## Brief changelog
tag with multi level will search tag as following: 
requestTag=beta|team1|partner1
step0: if force=true , just search tag(beta|team1|partner1)
step1:  search invokers with tag(beta|team1|partner1) , if result is empty , run step2
step2: search invokers with tag(beta|team1), if result is empty , run step3
step3: search invokers with tag(beta)

----
there are 10 deploy with tag=null and 10 deploy with dubbo.tag=beta, but my team just work with 8 deploy for a big feature-A , and my partners just work with 3 deploy for a small feature-A-1 :

my team want deploy 8 deploy with tag=beta|team1 , and share the deploy(tag=beta)
my partners want deploy 3 deploy with tag=beta|team1|partner1 , and share the deploy(tag=beta|team1)

## Verifying this change
add some test case to search addresses with muti level tag

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by AlbumenJ — APPROVED** [review]
LGTM
@chickenlj PTAL

**Comment by carlvine500** [comment]
> @carlvine500 Please update the following documentation accordingly:
> 
> https://cn.dubbo.apache.org/zh-cn/overview/core-features/traffic/#%E6%A0%87%E7%AD%BE%E8%B7%AF%E7%94%B1%E8%A7%84%E5%88%99
> 
> https://cn.dubbo.apache.org/zh-cn/overview/core-features/traffic/tag-rule/
ok

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/tag/TagStateRouter.java [hunk 1]
```diff
@@ -306,4 +309,35 @@ public void stop() {
...
+     * @return
+     */
+    public static Set<String> selectAddressByTagLevel(Map<String, Set<String>> tagAddresses, String tagSelector, boolean isForce) {
+        if (isForce || StringUtils.isNotContains(tagSelector, TAG_SEPERATOR)) {
+            return tagAddresses.get(tagSelector);
+        }
+        String[] selectors = StringUtils.split(tagSelector, TAG_SEPERATOR);
+        for (int i = selectors.length; i > 0; i--) {
+            String selectorTmp = StringUtils.join(selectors, TAG_SEPERATOR, 0, i);
+            System.out.println(selectorTmp);
```
Comment (wxbty) [inline dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/tag/TagStateRouter.java:334]: please remove System.out.println

File: dubbo-dependencies-bom/pom.xml [hunk 2]
```diff
@@ -96,7 +96,7 @@
         <javassist_version>3.29.2-GA</javassist_version>
         <bytebuddy.version>1.14.5</bytebuddy.version>
         <netty_version>3.2.10.Final</netty_version>
-        <netty4_version>4.1.92.Final</netty4_version>
+        <netty4_version>4.1.94.Final</netty4_version>
```
Comment (wxbty) [inline dubbo-dependencies-bom/pom.xml:99]: Why change the version?

**Review by wxbty — CHANGES_REQUESTED** [review]

File: dubbo-dependencies-bom/pom.xml (hunk 2 above)
Comment (carlvine500) [reply dubbo-dependencies-bom/pom.xml:99]: > Why change the version?

4.1.92.Final has vulnerability ,  CVE-2023-34462. Severity: MEDIUM
refer : https://mvnrepository.com/artifact/io.netty/netty-handler

I will rollback it , hope dubbo community will upgrade it later .

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/tag/TagStateRouter.java (hunk 1 above)
Comment (carlvine500) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/tag/TagStateRouter.java:334]: > please remove System.out.println

ok

**Review by wxbty — APPROVED** [review]
LGTM
btw, please resolve conflicts with the master

**Comment by AlbumenJ** [comment]
> > LGTM btw, please resolve conflicts with the master
> 
> I have no permission to merge to (apache:3.3)[apache:3.3](https://github.com/apache/dubbo/tree/3.3) , but I have tried merge to my git : https://github.com/carlvine500/dubbo/commits/3.3

Merge apache:3.3 into **your** branch

**Comment by carlvine500** [comment]
@AlbumenJ  failing test is not about this pull_request  .

**Review by AlbumenJ — COMMENTED** [review]
@chickenlj PTAL

**Pull request merged**